src/plugins/intel_cpu/tests/unit/vectorized/paged_attn_cache_rotation.cpp

```diff
@@ -0,0 +1,506 @@
+                                     raw_rotation_coefficients_mem_ptr,
+                                     num_heads,
+                                     block_size,
+                                     embedding_size);
+        }
+
+        auto ref_values_after_rotation = get_matrix_from_mem(cache_block_mem_ref, num_heads, block_size, embedding_size);
+        auto opt_values_after_rotation = get_matrix_from_mem(cache_block_mem_hw, num_heads, block_size, embedding_size);
+        compare_with_tolerance(opt_values_after_rotation, ref_values_after_rotation, get_tolerance<TypeParam>());
+    }
+};
+
+using OV_FP_TYPES = ::testing::Types<float, ov::float16, ov::bfloat16>;
+
+TYPED_TEST_SUITE_P(CacheRotationKernelInputTypeParameterizedTest);
+
+TYPED_TEST_P(CacheRotationKernelInputTypeParameterizedTest, RefBlockRotationGivesReferenceResults) {
+    auto raw_cache_mem_ptr = this->cache_mem.data();
+    auto raw_rotation_coefficients_mem_ptr = this->rotation_coefficients_mem.data();
+
+    rotate_kv_cache_block_ref(raw_cache_mem_ptr,
+                             raw_rotation_coefficients_mem_ptr,
+                             this->num_heads,
+                             this->block_size,
+                             this->embedding_size);
+
+    auto test_values_after_rotation =
+        get_matrix_from_mem(this->cache_mem, this->num_heads, this->block_size, this->embedding_size);
+    compare_with_tolerance(test_values_after_rotation, this->ref_values_after_rotation, get_tolerance<TypeParam>());
+}
+
+enum class TargetInstructionSet { AVX2, AVX512 };
+
+#if defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wsuggest-override" // false positive in gtest macro internals
+#endif
+
+MATCHER_P3(IsNFirstValuesNear, ref_container, abs_err, n, "") {
+    if (ref_container.size() < n || arg.size() < n)
+        return false;
+    if (ref_container.size() != arg.size())
+        return false;
+
+    bool is_ok = true;
+    for (size_t i = 0; i < n; i++) {
+        if (!::testing::ExplainMatchResult(::testing::FloatNear(static_cast<float>(arg[i]), abs_err),
+                                           static_cast<float>(ref_container[i]),
+                                           result_listener)) {
+            *result_listener << " for element at idx " << i << '\n';
+            is_ok = false;
+        }
+    }
+    return is_ok;
+}
+
+
+#if defined(__GNUC__)
+#pragma GCC diagnostic pop
+#endif
+
+class CacheRotationKernelInstructionParameterizedTest
+    : public ::testing::TestWithParam<std::tuple<TargetInstructionSet, size_t>> {
+protected:
+    constexpr static size_t MAX_CHUNK_SIZE_IN_ELEMENTS = 16;
+    template <class T>
+    using MemChunk = std::array<T, MAX_CHUNK_SIZE_IN_ELEMENTS>;
+
+    template <class T>
+    void test_chunk_rotation_for_type() {
+        auto instruction_set = std::get<0>(GetParam());
+        if (instruction_set == TargetInstructionSet::AVX512 && (!ov::with_cpu_x86_avx512f())) {
```

Comment:
We need the same condition for avx2. We still officially support CPUs with SSE isa only (even though we don't have them in pre-commit) - need to keep the tests green there.

Reply:
Done